src/vmm/src/core/vmm.rs

```diff
@@ -277,15 +340,63 @@ impl VMM {
         kernel_path: &Path,
         initramfs_path: &Option<PathBuf>,
     ) -> Result<()> {
+        let cmdline_extra_parameters = &mut Vec::new();
+
         self.configure_memory(mem_size_mb)?;
+        self.configure_allocators(mem_size_mb)?;
+        self.configure_net_device(cmdline_extra_parameters)?;
+
         let kernel_load = kernel::kernel_setup(
             &self.guest_memory,
             kernel_path.to_path_buf(),
             initramfs_path.clone(),
+            cmdline_extra_parameters,
         )?;
         self.configure_io()?;
         self.configure_vcpus(num_vcpus, kernel_load)?;
 
         Ok(())
     }
+
+    pub fn configure_net_device(
+        &mut self,
+        cmdline_extra_parameters: &mut Vec<String>,
+    ) -> Result<()> {
+        let mem = Arc::new(self.guest_memory.clone());
+        let range = if let Some(allocator) = &self.address_allocator {
+            allocator
+                .to_owned()
+                .allocate(0x1000, DEFAULT_ADDRESS_ALIGNEMNT, DEFAULT_ALLOC_POLICY)
+                .map_err(Error::Allocate)?
+        } else {
+            // Handle the case where self.address_allocator is None
+            panic!("Address allocator is not initialized");
+        };
+        let mmio_range = MmioRange::new(MmioAddress(range.start()), range.len())
+            .map_err(|_| Error::MmioRange)?;
+        let irq = self.irq_allocator.next_irq().map_err(Error::IrqAllocator)?;
+        let mmio_cfg = MmioConfig {
+            range: mmio_range,
+            gsi: irq,
+        };
+        // let mut guard = self.device_mgr.lock().unwrap();
+
+        // !TODO: MMIO Device Discovery + MMIO Device Register Layout
+        let net = Net::new(
```

Comment:
Those can be removed.
```suggestion
        };

        let net = Net::new(
```

Reply:
Done